### PR TITLE
Refactor descriptions

### LINKED ISSUE #167: Unexpected verbose output for `ops.help("create")`
The improvements made to `ops.help()` have been great for helping me explore the new API. But there is some unexpected behavior like `ops.help("create")`.

For example here's the output for `deconvolve`:

```python
print(ops.help("deconvolve"))
Names:
	> deconvolve.accelerate
	> deconvolve.firstGuess
	> deconvolve.normalizationFactor
	> deconvolve.richardsonLucy
	> deconvolve.richardsonLucyCorrection
	> deconvolve.richardsonLucyTV
	> deconvolve.richardsonLucyUpdate
```

But when I do the same for `create` I get this:

```python
print(ops.help("create"))
Ops:
	> create(
		 Inputs:
			Dimensions null
			Object null
		 Outputs:
			ImgFactory<Object> output1 -> imgFactory
	)
	Aliases: [create.imgFactory]
	
	> create(
		 Inputs:
			Dimensions null
			Type null
		 Outputs:
			Img<Type> output1 -> img
	)
	Aliases: [create.img]
	
	> create(
		 Inputs:
			Dimensions null
			Type null
			ImgFactory<Type> null
		 Outputs:
			Img<Type> output1 -> img
	)
	Aliases: [create.img]
	
	> create(
		 Inputs:
			Dimensions null
			IntegerType null
		 Outputs:
			ImgLabeling<Object, IntegerType> output1 -> imgLabeling
	)
	Aliases: [create.imgLabeling]
	
	> create(
		 Inputs:
			Dimensions null
			IntegerType null
			ImgFactory<IntegerType> null
		 Outputs:
			ImgLabeling<Object, IntegerType> output1 -> imgLabeling
	)
	Aliases: [create.imgLabeling]
	
	> create(
		 Inputs:
			double[][] null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> kernelRAI
	)
	Aliases: [create.kernel]
	
	> create(
		 Inputs:
			double[] null
			Integer null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> biGaussKernelRAI
	)
	Aliases: [create.kernel2ndDerivBiGauss]
	
	> create(
		 Inputs:
			double[] null
			Integer null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> biGaussKernelRAI
	)
	Aliases: [create.kernelBiGauss]
	
	> create(
		 Inputs:
			Dimensions null
			Double null
			Double null
			Double null
			Double null
			Double null
			Double null
			Double null
			ComplexType+NativeType null
		 Outputs:
			Img<ComplexType+NativeType> output1 -> diffractionKernelRAI
	)
	Aliases: [create.kernelDiffraction]
	
	> create(
		 Inputs:
			double[] null
			double[] null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> gaborKernelRAI
	)
	Aliases: [create.kernelGabor]
	
	> create(
		 Inputs:
			Double null
			double[] null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> gaborKernelRAI
	)
	Aliases: [create.kernelGabor]
	
	> create(
		 Inputs:
			double[] null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> gaussKernelRAI
	)
	Aliases: [create.kernelGauss]
	
	> create(
		 Inputs:
			Double null
			Integer null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> gaussKernelRAI
	)
	Aliases: [create.kernelGauss]
	
	> create(
		 Inputs:
			double[] null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> logKernelRAI
	)
	Aliases: [create.kernelLog]
	
	> create(
		 Inputs:
			Double null
			Integer null
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> logKernelRAI
	)
	Aliases: [create.kernelLog]
	
	> create(
		 Inputs:
			ComplexType null
		 Outputs:
			RandomAccessibleInterval<ComplexType> output1 -> sobelKernelRAI
	)
	Aliases: [create.kernelSobel]
	
	> create(
		 Inputs:
			Type null
		 Outputs:
			Type output1 -> type
	)
	Aliases: [create.type]
	
	> create(
		 Inputs:
		 Outputs:
			BitType output1 -> output
	)
	Aliases: [create.bit]
	
	> create(
		 Inputs:
		 Outputs:
			ByteType output1 -> output
	)
	Aliases: [create.byte]
	
	> create(
		 Inputs:
		 Outputs:
			ComplexFloatType output1 -> output
	)
	Aliases: [create.cfloat32]
	
	> create(
		 Inputs:
		 Outputs:
			ComplexDoubleType output1 -> output
	)
	Aliases: [create.cfloat64]
	
	> create(
		 Inputs:
		 Outputs:
			FloatType output1 -> output
	)
	Aliases: [create.float32]
	
	> create(
		 Inputs:
		 Outputs:
			DoubleType output1 -> output
	)
	Aliases: [create.float64]
	
	> create(
		 Inputs:
		 Outputs:
			IntType output1 -> output
	)
	Aliases: [create.int32]
	
	> create(
		 Inputs:
		 Outputs:
			LongType output1 -> output
	)
	Aliases: [create.int64]
	
	> create(
		 Inputs:
		 Outputs:
			ShortType output1 -> output
	)
...
```

I expected an output like this:

```python
	> create.Img
	> create.ImgFactory
	> create.ImgLabeling
	> create.ImgPlus
	> create.kernel
	> create.kernelDiffraction
	> create.kernelGabor
...
```

There are a few other namespaces that do this, which I think is okay if there are no sub methods (_i.e._ `namespace.method`) but it didn't make sense to me in the `create` namespace case.

### PR DESCRIPTION
**Pull request by gselzer**
This PR is (currently) a messy attempt at making the return from `DefaultOpEnvironment.help()` a lot cleaner to read. We make the following broad changes:
* The multiline description is condensed into a single line
* The only name shown is the searched name - no aliases are shown, and I can't think of a reason you'd want to know the aliases for an Op.
* All Op names are required to belong to a namespace. `engine.adapt` Ops now use the `engine.create` and `engine.copy` names where necessary for adaptation purposes. Other Ops were altered as needed to enforce this requirement.
* Argument types are now omitted, and only the names are shown.
* Container and Mutable arguments are now shown using the symbols `*` and `^`, respectively. @ctrueden liked this the best.

Note that this PR is designed to close scijava/scijava#167, and is meant to give @hinerm a starting point since he assigned himself that issue.

Points for further discussion:
* Using the names, and not the types, makes collapsing `OpInfo`s harder, because right now we only collapse based on equal `String`s. But, if the names are scraped from javadoc, we can't stop someone from e.g. making a typo and adding a new entry to the help.
* Do we think that the new key for container and mutable outputs are visible enough?
* ~~I never changed the verbose description - how should that change, if at all?~~ With @elevans' help, I edited the verbose descriptions to look similar to the simple descriptions, but they also include type/implementation/description details.

TODO:
* [ ] Once we finalize the design, update the docs!

### CODE REVIEW

File: scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/impl/OpWrappers.java [hunk 1]
```diff
@@ -19,11 +19,7 @@
 import org.scijava.ops.engine.OpWrapper;
 import org.scijava.ops.api.RichOp;
 
-public final class OpWrappers {
-
-	private OpWrappers() {
-		// Prevent instantiation of static utility class
-	}
```
Comment (gselzer) [inline scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/impl/OpWrappers.java:26]: Was this not a part of the `OpWrappers.vm` file?

File: scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/simplify/SimplifiedOpDescriptionGenerator.java [hunk 2]
```diff
@@ -63,12 +65,13 @@ public String simpleDescriptions(OpEnvironment env, OpRequest req) {
 		var infos = SimplificationMatchingRoutine.getInfos(env, req.getName());
 		var filtered = filterInfos(infos, req);
 		String opString = filtered.stream() //
-				.map(Infos::describe) //
```
Comment (gselzer) [inline scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/simplify/SimplifiedOpDescriptionGenerator.java:66]: We should consider deleting unused methods - there are now 4 different `Infos.describe...` methods - decide whether we need/want all of them

File: scijava-ops-engine/src/main/java/org/scijava/ops/engine/util/Infos.java [hunk 3]
```diff
@@ -137,17 +146,39 @@ public static List<OpDependencyMember<?>> dependencies(OpInfo info) {
...
-		return describe(info, null);
+		return describe(info, null, null);
+	}
+
+	/**
+	 * Generates a {@link String} describing the given {@link OpInfo}
+	 *
+	 * @param info the {@link OpInfo} of interest
+	 * @param name the name that the Op was searched by
+	 * @return a descriptor for {@code info}
```
Comment (gselzer) [inline scijava-ops-engine/src/main/java/org/scijava/ops/engine/util/Infos.java:157]: Again, figure out how much of this is needed

File: scijava-ops-engine/src/test/java/org/scijava/ops/engine/matcher/impl/OpDescriptionTest.java [hunk 4]
```diff
@@ -59,7 +59,7 @@ public void testOpClassDescription() {
 
 		String expected = "test.classDescription(\n\t " //
 			+ "Inputs:\n\t\tjava.lang.Double input1\n\t\tjava.lang.Double input2\n\t " //
-			+ "Outputs:\n\t\tjava.lang.Double output1\n)\n";
+			+ "Output:\n\t\tjava.lang.Double output1\n)\n";
 		String actual = info.toString();
 		Assertions.assertEquals(expected, actual);
 	}
```
Comment (gselzer) [inline scijava-ops-engine/src/test/java/org/scijava/ops/engine/matcher/impl/OpDescriptionTest.java:65]: We may want to switch `info.toString()` to the current `Infos.describeMultiline` if we like that better

File: scijava-ops-engine/templates/main/java/module-info.vm [hunk 5]
```diff
@@ -134,7 +134,8 @@ module org.scijava.ops.engine {
 	    org.scijava.ops.engine.math.Zero,
 	    org.scijava.ops.engine.matcher.simplify.PrimitiveArraySimplifiers,
 	    org.scijava.ops.engine.matcher.simplify.PrimitiveSimplifiers,
-		org.scijava.ops.engine.stats.Size;
+			org.scijava.ops.engine.stats.Mean,
+			org.scijava.ops.engine.stats.Size;
```
Comment (gselzer) [inline scijava-ops-engine/templates/main/java/module-info.vm:138]: Why is this indented

File: scijava-ops-image/src/main/java/org/scijava/ops/image/convert/Converters.java [hunk 6]
```diff
@@ -42,7 +42,7 @@ public class Converters<I, O extends Type<O>> {
 	 * @input converter
 	 * @input outputType
 	 * @output outputII
-	 * @implNote op names='convert'
+	 * @implNote op names='convert.image'
```
Comment (gselzer) [inline scijava-ops-image/src/main/java/org/scijava/ops/image/convert/Converters.java:45]: Some Ops use `img`, others `image` - we should probably be consistent

File: scijava-ops-image/src/main/java/org/scijava/ops/image/filter/tubeness/DefaultTubeness.java [hunk 7]
```diff
@@ -99,7 +99,7 @@ public class DefaultTubeness<T extends RealType<T>> implements
 	private Function<Dimensions, ImgFactory<DoubleType>> createFactoryOp;
 
 	//TODO: make sure this works
```
Comment (gselzer) [inline scijava-ops-image/src/main/java/org/scijava/ops/image/filter/tubeness/DefaultTubeness.java:101]: Does it work?

File: scijava-ops-image/src/main/java/org/scijava/ops/image/transform/slice/slice/SlicesII.java [hunk 8]
```diff
@@ -27,7 +27,7 @@
  * #L%
  */
 
-package org.scijava.ops.image.slice;
+package org.scijava.ops.image.transform.slice.slice;
```
Comment (gselzer) [inline scijava-ops-image/src/main/java/org/scijava/ops/image/transform/slice/slice/SlicesII.java:30]: Why is this package `slice.slice`?

File: scijava-ops-image/src/test/java/org/scijava/ops/image/create/CreateNativeTypeTest.java [hunk 9]
```diff
@@ -54,11 +54,11 @@ public class CreateNativeTypeTest extends AbstractOpTest {
 	@Test
 	public <T extends NativeType<T>> void testCreateNativeType() {
 
-		Producer<NativeType> typeSource = ops.nullary("create") //
+		Producer<NativeType> typeSource = ops.nullary("create.type") //
 				.outType(NativeType.class) //
 				.producer();
 
-		Function<Class<T>, T> typeFunc = ops.unary("create") //
+		Function<Class<T>, T> typeFunc = ops.unary("create.object") //
```
Comment (gselzer) [inline scijava-ops-image/src/test/java/org/scijava/ops/image/create/CreateNativeTypeTest.java:61]: Consider coming up with a better name...

File: scijava-ops-image/src/test/java/org/scijava/ops/image/transform/project/project/ProjectTest.java [hunk 10]
```diff
@@ -27,7 +27,7 @@
  * #L%
  */
 
-package org.scijava.ops.image.project;
+package org.scijava.ops.image.transform.project.project;
```
Comment (gselzer) [inline scijava-ops-image/src/test/java/org/scijava/ops/image/transform/project/project/ProjectTest.java:30]: This package should not have two `project`

File: scijava-types/src/test/java/org/scijava/types/inference/InferTypeVariablesTest.java [hunk 11]
```diff
@@ -407,4 +407,5 @@ public <T> void testWildcardTypeInference() throws TypeInferenceException {
 		Assertions.assertEquals(expected2, typeAssigns2);
 	}
 
+
```
Comment (gselzer) [inline scijava-types/src/test/java/org/scijava/types/inference/InferTypeVariablesTest.java:410]: Can remove this

**Review by gselzer — COMMENTED** [review]
I like the changes, but as expected with a draft PR it's sloppy

**Review by hinerm — APPROVED** [review]

**Pull request merged**